### PR TITLE
[WIP] Run distributed tree constructor on the device

### PR DESCRIPTION
**Pull request by masterleinad**
This is to separate the (fundamental) CUDA-aware MPI changes from the changes to the constructor (commits 10104ff  and eb2fa93). These two last commits expose some strange behavior to look at later. We observe some issues with floatring point issues (partly addressed by 10104ff) and success of one of the tests (`one_leaf_per_rank`) depends on not running `hello_world`.

### CODE REVIEW

**Comment by masterleinad** [comment]
This is now rebased on top of `master` so all changes are directly related to this pull request.

**Comment by masterleinad** [comment]
I checked the performance impact on `Summit` and don't really see anything significant. The changes in the timings for the construction are contained in the fluctuations I see anyway.

File: src/ArborX_DistributedSearchTree.hpp [hunk 1]
```diff
@@ -121,14 +121,28 @@ DistributedSearchTree<DeviceType>::DistributedSearchTree(
   Kokkos::View<Box *, DeviceType> boxes(
       Kokkos::ViewAllocateWithoutInitializing("rank_bounding_boxes"),
       comm_size);
-  // FIXME when we move to MPI with CUDA-aware support, we will not need to
-  // copy from the device to the host
+#ifdef ARBORX_USE_CUDA_AWARE_MPI
+  if (!_bottom_tree.empty())
+  {
+    Kokkos::View<Box const, DeviceType, Kokkos::MemoryTraits<Kokkos::Unmanaged>>
+        root_bounding_volume(&this->getBoundingVolume(this->getRoot()));
```
Comment (masterleinad) [inline src/ArborX_DistributedSearchTree.hpp]: Yes, fascinating that `_bottom_tree.getBoundingVolume` doesnt work here. At least the compiler is very clear about it. :slightly_smiling_face:

File: src/ArborX_DistributedSearchTree.hpp [hunk 2]
```diff
@@ -97,8 +97,8 @@ class DistributedSearchTree
 private:
   friend struct Details::DistributedSearchTreeImpl<DeviceType>;
   MPI_Comm _comm;
-  BVH<DeviceType> _top_tree;    // replicated
-  BVH<DeviceType> _bottom_tree; // local
+  BVH<DeviceType> _top_tree;             // replicated
+  BVH<DeviceType> &_bottom_tree = *this; // local
```
Comment (masterleinad) [inline src/ArborX_DistributedSearchTree.hpp]:
```suggestion
  BVH<DeviceType> const &_bottom_tree = *this; // local
```

**Comment by aprokop** [comment]
The cleaned up version was merged in #597. Closing.

**Pull request closed without merging**